src/www/ui_components/language-view.html

```diff
@@ -0,0 +1,93 @@
+<!--
+  Copyright 2018 The Outline Authors
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+<link rel="import" href="../bower_components/polymer/polymer-element.html" />
+<link rel="import" href="../bower_components/polymer/lib/elements/dom-repeat.html" />
+<link rel="import" href="../bower_components/polymer/lib/elements/dom-if.html" />
+
+<dom-module id="language-view">
+  <template>
+    <style>
+      :host {
+        background: #fff;
+        display: flex;
+        flex-direction: column;
+        justify-content: space-between;
+        text-align: center;
+        width: 100%;
+        height: 100vh;
+      }
+      .language-item {
+        display: flex;
+        cursor: pointer;
+        font-size: 16px;
+        border-bottom: 1px solid #e0e0e0;
+        padding-left: 24px;
+        --paper-item-selected: {
+          color: var(--medium-green);
+          font-weight: normal;
+        }
+      }
+      .language-name {
+        text-align: left;
+        flex-grow: 1;
+      }
+    </style>
+
+    <div id="main">
+      <paper-listbox selected="{{language}}" attr-for-selected="value" on-selected-changed="_languageSelected">
+        <template is="dom-repeat" items="{{languages}}" as="lang" sort="_sortLanguageNameAscending">
+          <paper-item class="language-item" value="{{lang.id}}">
+            <span class="language-name">{{lang.name}}</span>
+            <iron-icon icon="check" hidden$="{{_shouldHideCheck(language, lang.id)}}"></iron-icon>
+          </paper-item>
+        </template>
+      </paper-listbox>
+    </div>
+  </template>
+
+  <script>
+    "use strict";
+
+    Polymer({
+      is: "language-view",
+      properties: {
+        language: {
+          type: String,
+        },
+        // An array of {id, name, dir} language objects.
+        languages: {
+          type: Array,
+          readonly: true,
+        },
+      },
+      _sortLanguageNameAscending(a, b) {
+        return a.name > b.name ? 1 : -1;
+      },
+      _languageSelected(event) {
+        const languageCode = event.detail.value;
+        this.fire("ChangeLanguage", {languageCode});
+        this.language = languageCode;
+
+        // TODO: How to properly go back to the main screen?
```

Comment:
I'm pretty sure there is a better way to do this. Can someone help?

Reply:
I was thinking of something like https://ionicframework.com/docs/api/modal

Reply:
app-root has a method to [go back](https://github.com/Jigsaw-Code/outline-client/blob/master/src/www/ui_components/app-root.html#L672). You could delegate closing to the parent element.

Reply:
Thanks! I updated it to fire an event and use app.ts#changeToDefaultPage to go back.